torch/distributed/optim/post_localSGD_optimizer.py

```diff
@@ -65,10 +66,19 @@ def __repr__(self):
         return self.optim.__repr__()
 
     def state_dict(self):
-        return self.optim.state_dict()
+        optim_state_dict = self.optim.state_dict()
+        if optim_state_dict is not None and self.averager.step is not None:
```

Comment:
There is no need to check that `optim_state_dict is not None` since `torch.optim.Optimizer.state_dict()` never returns `None`.

Similarly, for `self.averager.step`, I do not believe it can be `None`.

Reply:
got it, removed